### PR TITLE
Fix wrong sidebar style on mobile

### PR DESCRIPTION
**Pull request by stevenjoezhang**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes was maked (for bug fixes / features).
   - [x] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [ ] Docs in [NexT website](https://theme-next.org/docs/) have been added / updated (for new features).

## PR Type
**What kind of change does this PR introduce?**

- [x] Bugfix.
- [ ] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

Issue resolved: #807

## What is the new behavior?

- Screenshots with this changes: N/A
- Link to demo site with this changes: N/A

### How to use?
In NexT `_config.yml`:
```yml
...
```

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

File: source/css/_mixins/base.styl [hunk 1]
```diff
@@ -76,11 +76,6 @@ show() {
   display: block;
 }
 
-fixbutton() {
-  right: 20px;
-  opacity: 0.8;
-}
```
Comment (ivan-nginx) [inline source/css/_mixins/base.styl]: Toggle buttons (b2t too) have no `opacity` and `right` parameter. No need to remove this button.

**Comment by ivan-nginx** [comment]
Also, I wan't to create new issue for similar component and report here:

image

Color changed on hover b2t or sidebar toggle buttons. It's not must be (not was before).

**Comment by stevenjoezhang** [comment]
I think it's all fine. The old behaviour is even worse and confusing users.

File: source/css/_mixins/base.styl (hunk 1 above)
Comment (stevenjoezhang) [reply source/css/_mixins/base.styl]: Then you will find b2t and sidebar button have different opacity and right-margin

**Comment by ivan-nginx** [comment]
What's fine? Bug fine or what?

**Comment by stevenjoezhang** [comment]
It's feature.

**Comment by stevenjoezhang** [comment]
It's better than before. So it's fine. That's simple.

**Comment by ivan-nginx** [comment]
> It's feature.

Then create a feature with this. No need to do excess additionals.

**Comment by ivan-nginx** [comment]
> It's better than before. So it's fine. That's simple.

Better or not – no your question. For now we have:

1. Bug reported in #807 and «fix» with additional bug (removed opacity & right align)
2. Added no-needed feature with hovering highlight. Highlight must be on posts, when TOC is exists.

File: source/css/_mixins/base.styl (hunk 1 above)
Comment (ivan-nginx) [reply source/css/_mixins/base.styl]: Are u sure in this?

Comment (ivan-nginx) [reply source/css/_mixins/base.styl]: image

Comment (ivan-nginx) [reply source/css/_mixins/base.styl]: image

This is your fix in this PR.

**Comment by stevenjoezhang** [comment]
So you want opacity: 0.8 on mobile?

**Comment by ivan-nginx** [comment]
https://github.com/theme-next/hexo-theme-next/pull/808/files#r272229828

**Comment by ivan-nginx** [comment]
I want no excess changes for anything. See? We get 2 bugs after your refactorings.

**Comment by stevenjoezhang** [comment]
Ok, ok, it's my fault. What about now?

**Comment by ivan-nginx** [comment]
Now right align needed. From fixbutton, how was before.

**Comment by stevenjoezhang** [comment]
About right margin, we only have one macro: `$b2t-position-right`, equal to 30px
Need another one like `$b2t-position-right-mobile`?

**Comment by ivan-nginx** [comment]
> Need another one like `$b2t-position-right-mobile`?

I think yes.

File: source/css/_variables/base.styl [hunk 2]
```diff
@@ -282,6 +282,7 @@ $b2t-opacity                  = 1
 $b2t-position-bottom          = -100px
```
Comment (ivan-nginx) [inline source/css/_variables/base.styl]: So, maybe we can use this `$b2t-opacity`? With the changed value to 0.8 or 0.7?

Comment (ivan-nginx) [reply source/css/_variables/base.styl]: Or not. Wait. There is on Pisces and Gemini opacity too...

**Comment by ivan-nginx** [comment]
About margins... I see with other options and other schemes positions different in other resolutions. It's not seems like bugs, but can be improved in the future.

**Comment by stevenjoezhang** [comment]
We can use `em` as unit of margin, like in https://github.com/theme-next/hexo-theme-next/pull/800

And, the font-size of back2top button could be larger in high-resolution screens.

**Comment by ivan-nginx** [comment]
> We can use `em` as unit of margin, like in #800
> 
> And, the font-size of back2top button could be larger in high-resolution screens.

Yep. But i think not now, are u agree? At least after fonts refactoring.
And your comment can be posted as issue with low priority, for example.

**Comment by stevenjoezhang** [comment]
Agree. Needs to be done in the future.

**Pull request merged**